client/src/components/SurveyStepper.tsx

```diff
@@ -316,18 +317,29 @@ export default function SurveyStepper({
           connector={null}
         >
           {survey.pages.map((page, index) => (
-            <Step key={page.id} completed={false}>
+            <Step key={page.id} completed={index < pageNumber ? true : false}>
```

Comment:
Ternary operation not needed. 
```suggestion
            <Step key={page.id} completed={index < pageNumber}>
```